packages/terra-modal/src/Modal.jsx

```diff
@@ -0,0 +1,142 @@
+import React, { PropTypes } from 'react';
+import classNames from 'classnames';
+import Portal from 'react-portal';
+import ModalContent from './ModalContent';
+import './Modal.scss';
+
+const propTypes = {
+  /**
+   * String that labels the modal for screen readers
+   **/
+  ariaLabel: PropTypes.string.isRequired,
+  /**
+   * This callback is called when the closing event is triggered but it prevents normal removal from the DOM. So, you can do some DOMNode animation first and then call removeFromDOM() that removes the modal from DOM.
+   **/
+  beforeClose: PropTypes.func,
+  /**
+   * Content inside the modal dialog
+   **/
+  children: PropTypes.element.isRequired,
```

Comment:
Similar to how we standardized the height / width properties I think children is also something we should standardize.

I believe most components specify the child type as node so that any renderable object can be passed into children. I think it becomes a little confusing when some components can render anything while other components will only accept element types.